wasm/src/main/java/com/dylibso/chicory/wasm/types/NameCustomSection.java

```diff
@@ -12,30 +12,59 @@
  */
 public class NameCustomSection extends CustomSection {
 
-    private String moduleName;
-    private final ArrayList<NameEntry> funcNames = new ArrayList<>();
-    private final ArrayList<ListEntry<NameEntry>> localNames = new ArrayList<>();
-    private final ArrayList<ListEntry<NameEntry>> labelNames = new ArrayList<>();
-    private final ArrayList<NameEntry> tableNames = new ArrayList<>();
-    private final ArrayList<NameEntry> memoryNames = new ArrayList<>();
-    private final ArrayList<NameEntry> globalNames = new ArrayList<>();
-    private final ArrayList<NameEntry> elementNames = new ArrayList<>();
-    private final ArrayList<NameEntry> dataNames = new ArrayList<>();
-    private final ArrayList<NameEntry> tagNames = new ArrayList<>();
+    private final String moduleName;
+    private final ArrayList<NameEntry> funcNames;
+    private final ArrayList<ListEntry<NameEntry>> localNames;
+    private final ArrayList<ListEntry<NameEntry>> labelNames;
+    private final ArrayList<NameEntry> tableNames;
+    private final ArrayList<NameEntry> memoryNames;
+    private final ArrayList<NameEntry> globalNames;
+    private final ArrayList<NameEntry> elementNames;
+    private final ArrayList<NameEntry> dataNames;
+    private final ArrayList<NameEntry> tagNames;
 
     /**
      * Construct a new, empty section instance.
      */
-    public NameCustomSection() {}
+    public NameCustomSection(
+            String moduleName,
+            ArrayList<NameEntry> funcNames,
```

Comment:
If we make these `List<X>` rather than `ArrayList<X>`, then the caller can pass in any list type. We should also make defensive copies, to ensure that the class is immutable:
```java
this.moduleName = List.copyOf(moduleName);

Reply:
+1

Reply:
good catch, thanks :+1: